templates/connect-inject-deployment.yaml

```diff
@@ -41,6 +41,23 @@ spec:
               valueFrom:
                 fieldRef:
                   fieldPath: metadata.namespace
+            {{- /* An ACL token is only necessary for the connect injector if namespaces are enabled */}}
+            {{- if .Values.global.consulNamespacesEnabled }}
+            {{- if (and .Values.connectInject.aclInjectToken.secretName .Values.connectInject.aclInjectToken.secretKey) }}
+            - name: CONSUL_HTTP_TOKEN
+              valueFrom:
+                secretKeyRef:
+                  name: {{ .Values.connectInject.aclInjectToken.secretName }}
+                  key: {{ .Values.connectInject.aclInjectToken.secretKey }}
+            {{- end }}
+            {{- if .Values.global.bootstrapACLs }}
```

Comment:
should this be an else if? What happens if the env var is set twice?

Reply:
Updated.